### PR TITLE
Limit and LimitWeighted operator docs

### PR DESCRIPTION
**Pull request by johanandren**
Refs #25468

### CODE REVIEW

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limit.md [hunk 1]
```diff
@@ -4,17 +4,27 @@ Limit number of element from upstream to given `max` number.
...
 
-@@signature [Flow.scala](/akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala) { #limit }
-
-@@@
+@apidoc[Flow.limit](Flow$) { scala="#limit(max:Long):FlowOps.this.Repr[Out]" java="#limit(long)" } 
 
 ## Description
 
-Limit number of element from upstream to given `max` number.
+Limit number of element from upstream to given `max` number, if the limit is passed the operator fails the stream with a @apidoc[StreamLimitReachedException](StreamLimitReachedException).
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limit.md]:
```suggestion
Limits the number of elements from upstream to a given `max` number, if the limit is passed the operator fails the stream with a @apidoc[StreamLimitReachedException](StreamLimitReachedException).
```

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limit.md [hunk 2]
```diff
@@ -4,17 +4,27 @@ Limit number of element from upstream to given `max` number.
...
+
+See also @ref:[limitWeighted](limitWeighted.md) which can choose a weight for each element counting to a total max limit weight. @ref:[take](take.md) is also closely related but completes the stream instead of failing it after a certain number of elements.
+
+## Example
+
+`limit` can protect a stream coming from an untrusted source into an in memory aggregate that grows with the number of element from filling the heap and causing an OOM.
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limit.md]:
```suggestion
`limit` can protect a stream coming from an untrusted source into an in-memory aggregate that grows with the number of elements from filling the heap and causing an out-of-memory error.
```

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limit.md [hunk 3]
```diff
@@ -4,17 +4,27 @@ Limit number of element from upstream to given `max` number.
...
+In this sample we take at most 10 000 from the untrusted source elements into the aggregated sequence of elements, if the untrusted source emits more elements the stream and the materialized @scala[`Future[Seq[String]]`]@java[`CompletionStage<List<String>>`] will be failed:
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limit.md]:
```suggestion
In this sample we take at most 10 000 of the untrusted source elements into the aggregated sequence of elements, if the untrusted source emits more elements the stream and the materialized @scala[`Future[Seq[String]]`]@java[`CompletionStage<List<String>>`] will be failed:
```

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limitWeighted.md [hunk 4]
```diff
@@ -1,21 +1,30 @@
...
-@@@
+@apidoc[Flow.limitWeighted](Flow) { scala="#limitWeighted[T](max:Long)(costFn:Out=&gt;Long):FlowOps.this.Repr[Out]" java="#limitWeighted(long,akka.japi.function.Function)" } 
 
 ## Description
 
-Ensure stream boundedness by evaluating the cost of incoming elements using a cost function.
-Evaluated cost of each element defines how many elements will be allowed to travel downstream.
+A weight function returns the weight of each element, then the total accumulated weight  is compared to a max and if it has passed the max the stream is failed with a @apidoc[StreamLimitReachedException](StreamLimitReachedException).
+
+See also @ref:[limit](limit.md) which puts a number on the number of elements instead (the same as always returning `1` from the weight function).
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limitWeighted.md]:
```suggestion
See also @ref:[limit](limit.md) which puts a limit on the number of elements instead (the same as always returning `1` from the weight function).
```

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limitWeighted.md [hunk 5]
```diff
@@ -1,21 +1,30 @@
...
+
+## Examples
+
+`limitWeighted` can protect a stream coming from an untrusted source into an in memory aggregate that grows with the number of element from filling the heap and causing an OOM.
```
Comment (ennru) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limitWeighted.md]:
```suggestion
`limitWeighted` can protect a stream coming from an untrusted source into an in-memory aggregate that grows with the number of elements from filling the heap and causing an out-of-memory error.
```

**Review by ennru — APPROVED** [review]
LGTM.

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limit.md [hunk 6]
```diff
@@ -4,17 +4,27 @@ Limit number of element from upstream to given `max` number.
...
-
-@@@
+@apidoc[Flow.limit](Flow$) { scala="#limit(max:Long):FlowOps.this.Repr[Out]" java="#limit(long)" } 
 
 ## Description
 
-Limit number of element from upstream to given `max` number.
+Limits the number of elements from upstream to a given `max` number, if the limit is passed the operator fails the stream with a @apidoc[StreamLimitReachedException](StreamLimitReachedException).
+
+See also @ref:[limitWeighted](limitWeighted.md) which can choose a weight for each element counting to a total max limit weight. @ref:[take](take.md) is also closely related but completes the stream instead of failing it after a certain number of elements.
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limit.md]: :+1: linking to take

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limitWeighted.md [hunk 7]
```diff
@@ -1,21 +1,30 @@
 # limitWeighted
 
-Ensure stream boundedness by evaluating the cost of incoming elements using a cost function.
+Limit a max total weight for incoming elements
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/limitWeighted.md]:
```suggestion
Limit the total weight of incoming elements
```
?

**Review by raboof — APPROVED** [review]
One tiny suggestion for consideration, LGTM with or without that change

**Comment by johanandren** [comment]
Argh, looks like I messed up somehow.

**Pull request merged**